feat(operator): add volcano scheduler support for grove (#8225)
ai-dynamo/dynamo: deploy/helm/charts/platform/README.md

| file://components/operator | dynamo-operator | 1.3.0 |
| https://charts.bitnami.com/bitnami | etcd | 12.0.18 |
| https://nats-io.github.io/k8s/helm/charts/ | nats | 1.3.2 |

> | oci://ghcr.io/ai-dynamo/grove | grove(grove-charts) | v0.1.0-alpha.9 |

Reviewer: so we need .10 apparently?

Author: Yes, let's wait for the Grove release team; they'll release this new version ASAP

|-----------------|---------------|-------|
| 1.0.x           | >= v0.13.0    | >= v0.1.0-alpha.6 |
| 1.1.x           | >= v0.13.4    | >= v0.1.0-alpha.8 |

> | 1.3.x           | >= v0.13.4    | >= v0.1.0-alpha.9 |

Reviewer: so we need .10 apparently?

Author: Yes, let's wait for the Grove release team; they'll release this new version ASAP

> | 1.3.x           | >= v0.13.4    | >= v0.1.0-alpha.10 |

Reviewer: we are now targetting 1.4

Reviewer: And with the changed binding resource, this means that actually 1.3 need <v0.1.0-alpha.10 ? When did clustertopology -> clustertopologybinding happen exactly? This deserved a note in the release too that the Grove version is crucial in 1.4.

Author: Done

|-----------------|---------------|-------|
| 1.0.x           | >= v0.13.0    | >= v0.1.0-alpha.6 |
| 1.1.x           | >= v0.13.4    | >= v0.1.0-alpha.8 |
| 1.3.x           | >= v0.13.4    | v0.1.0-alpha.8 |

> | 1.4.x           | >= v0.13.4    | v0.1.0-alpha.10 |

Reviewer: ```suggestion
| 1.3.x           | >= v0.13.4    | >= v0.1.0-alpha.8, < v0.1.0-alpha.9 (it's incompatible, see below) |
| 1.4.x           | >= v0.13.4    | >= v0.1.0-alpha.10 |
```
Does this make sense? We don't really know what comes with 10+.

Author: Done

> | 1.4.x           | >= v0.13.4    | v0.1.0-alpha.10 |

Reviewer: ```suggestion
| 1.3.x           | >= v0.13.4    | >= v0.1.0-alpha.8, < v0.1.0-alpha.9 (it's incompatible, see below) |
| 1.4.x           | >= v0.13.4    | >= v0.1.0-alpha.10 |
```
Does this make sense? We don't really know what comes with 10+.

Author: Done

> Grove must be upgraded in lockstep with Dynamo across Grove topology API changes. Dynamo 1.3.x expects Grove's earlier `ClusterTopology` API and should not be paired with Grove versions that renamed the resource to `ClusterTopologyBinding` (`v0.1.0-alpha.9` and newer). Dynamo 1.4.x expects the renamed `ClusterTopologyBinding` API and requires Grove `v0.1.0-alpha.10`, which also includes the Volcano scheduler backend support used by Dynamo's experimental Volcano integration.

Reviewer: ```suggestion
Note: It is recommended to upgrade Grove in lockstep with Dynamo as Grove's APIs are not considered stable yet. E.g. Dynamo 1.3.x expects Grove's earlier `ClusterTopology` API and cannot work with the newer `ClusterTopologyBinding`, Dynamo 1.4.x conversely.
```

Author: Done
